src/beast/beast/nudb/store.h

```diff
@@ -576,10 +568,22 @@ basic_store<Hasher, File>::insert (void const* key,
     {
         unique_lock_type m (m_);
         s_->p1.insert (h, key, data, size);
-        bool const full =
+        // Did we go over the commit limit?
+        if (commit_limit_ > 0 &&
+            s_->p1.data_size() >= commit_limit_)
+        {
+            // Yes, start a new commit
+            cond_.notify_all();
+            // Wait for pool to shrink
+            cond_limit_.wait(m,
+                [this]() { return
+                    s_->p1.data_size() <
+                        commit_limit_; });
```

Comment:
Well the thing is that data_size drops to zero instantly (when p1 is swapped with p0). It doesn't fall gently.